tests/pytests/unit/grains/test_core.py

```diff
@@ -3623,3 +3871,1154 @@ def _mock_is_file(filename):
+                }
+            ],
+            "inet6": [],
+        },
+        "eth1": {
+            "up": True,
+            "hwaddr": "00:00:00:00:00:00",
+            "inet": [
+                {
+                    "address": "0.0.0.0",
+                    "netmask": "255.255.255.0",
+                    "broadcast": "0.0.0.0",
+                    "label": "wlo1",
+                }
+            ],
+            "inet6": [],
+        },
+    }
+    with patch.object(core, "_get_interfaces", return_value=mock_get_interfaces):
+        ret = core.hwaddr_interfaces()
+        assert "hwaddr_interfaces" in ret
+        assert ret["hwaddr_interfaces"] == {
+            "lo": "00:00:00:00:00:00",
+            "eth1": "00:00:00:00:00:00",
+        }
+
+
+def test_id():
+    """
+    test id
+    """
+    ret = core.id_()
+    assert "id" in ret
+
+    with patch("salt.grains.core.__opts__", {"id": "test_id_minion_id"}):
+        ret = core.id_()
+        assert "id" in ret
+        assert ret["id"] == "test_id_minion_id"
+
+
+def test__linux_bin_exists():
+    """
+    test id
+    """
+    mock_retcode = [salt.exceptions.CommandExecutionError, 0]
+    with patch.dict(
+        core.__salt__, {"cmd.retcode": MagicMock(side_effect=mock_retcode)}
+    ):
+        ret = core._linux_bin_exists("ls")
+        assert ret
+
+    mock_retcode = salt.exceptions.CommandExecutionError
+    mock_runall = [
+        {"pid": 100, "retcode": 0, "stdout": "ls: /usr/bin/ls", "stderr": ""}
+    ]
+    with patch.dict(
+        core.__salt__, {"cmd.retcode": MagicMock(side_effect=mock_retcode)}
+    ):
+        with patch.dict(
+            core.__salt__, {"cmd.run_all": MagicMock(side_effect=mock_runall)}
+        ):
+            ret = core._linux_bin_exists("ls")
+            assert ret
+
+    mock_retcode = salt.exceptions.CommandExecutionError
+    mock_runall = salt.exceptions.CommandExecutionError
+
+    with patch.dict(
+        core.__salt__, {"cmd.retcode": MagicMock(side_effect=mock_retcode)}
+    ):
+        with patch.dict(
+            core.__salt__, {"cmd.run_all": MagicMock(side_effect=mock_runall)}
+        ):
+            ret = core._linux_bin_exists("ls")
+            assert not ret
+
+
+def test__parse_lsb_release():
+    """
+    test id
+    """
+    mock_lsb_file = """
+DISTRIB_ID="ManjaroLinux"
+DISTRIB_RELEASE="23.0.2"
+DISTRIB_CODENAME="Uranos"
+DISTRIB_DESCRIPTION="Manjaro Linux"
+"""
+
+    with patch("salt.utils.files.fopen", mock_open(read_data=mock_lsb_file)):
+        ret = core._parse_lsb_release()
+        assert ret == {
+            "lsb_distrib_id": "ManjaroLinux",
+            "lsb_distrib_release": "23.0.2",
+            "lsb_distrib_codename": "Uranos",
+            "lsb_distrib_description": "Manjaro Linux",
+        }
+
+    with patch("salt.utils.files.fopen", side_effect=OSError):
+        ret = core._parse_lsb_release()
+        assert ret == {}
+
+
+def test__osx_gpudata():
+    """
+    test id
+    """
```

Comment:
```suggestion
    """
    test id
    """
```
I believe these docstrings were just copy/pasted right?
Remove it?